src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.OpenFileTracker.cs

```diff
@@ -301,34 +364,35 @@ private void TryClosingDocumentsForMoniker(string moniker)
 
                 UnsubscribeFromWatchedHierarchies(moniker);
 
-                _workspace.ApplyChangeToWorkspace(w =>
-                {
-                    var documentIds = w.CurrentSolution.GetDocumentIdsWithFilePath(moniker);
-                    if (documentIds.IsDefaultOrEmpty)
+                _workspaceApplicationQueue.AddWork(() =>
+                    _workspace.ApplyChangeToWorkspaceMaybeAsync(useAsync: true, w =>
```

Comment:
Not sure if it's easier/better to just have the queue itself acquire the lock when it's running these changes; that would mean you could acquire once for the batch, which _might_ be better. Feel free to leave as is though.

Reply:
yeah, that's def a possibility.   let me see about doing that.

Reply:
ok.  i did that.  nice :)